ui/tests/acceptance/enterprise-kmip-test.js

```diff
@@ -0,0 +1,212 @@
+    await scopesPage.delete();
+    await scopesPage.confirmDelete();
+    assert.equal(scopesPage.listItemLinks.length, 0, 'renders a single scope');
+  });
+
+  test('it can create and delete a role', async function(assert) {
+    let { path, scope } = await createScope(this);
+    let role = `role-${Date.now()}`;
+    await rolesPage.visit({ backend: path, scope });
+    assert.ok(rolesPage.isEmpty, 'renders the empty role page');
+    await rolesPage.create();
+    assert.equal(
+      currentURL(),
+      `/vault/secrets/${path}/kmip/scopes/${scope}/roles/create`,
+      'links to the role create form'
+    );
+
+    await rolesPage.roleName(role);
+    await rolesPage.submit();
+    assert.equal(
+      currentURL(),
+      `/vault/secrets/${path}/kmip/scopes/${scope}/roles`,
+      'redirects to roles list'
+    );
+
+    assert.equal(rolesPage.listItemLinks.length, 1, 'renders a single role');
+
+    // delete the role
+    await rolesPage.listItemLinks.objectAt(0).menuToggle();
+    await rolesPage.delete();
+    await rolesPage.confirmDelete();
+    assert.equal(rolesPage.listItemLinks.length, 0, 'renders no roles');
+    assert.ok(rolesPage.isEmpty, 'renders empty');
+  });
+
+  test('it can delete a role from the detail page', async function(assert) {
+    let { path, scope, role } = await createRole(this);
+    await rolesPage.visitDetail({ backend: path, scope, role });
+    await rolesPage.detailEditLink();
+    assert.equal(
+      currentURL(),
+      `/vault/secrets/${path}/kmip/scopes/${scope}/roles/${role}/edit`,
+      'navigates to role edit'
+    );
+    await rolesPage.cancelLink();
+    assert.equal(
+      currentURL(),
+      `/vault/secrets/${path}/kmip/scopes/${scope}/roles/${role}`,
+      'cancel navigates to role show'
+    );
+    await rolesPage
+      .detailDelete()
+      .delete()
+      .confirmDelete();
+    assert.equal(
+      currentURL(),
+      `/vault/secrets/${path}/kmip/scopes/${scope}/roles`,
+      'redirects to the roles list'
+    );
+    assert.ok(rolesPage.isEmpty, 'renders an empty roles page');
+  });
+
+  test('it can create and delete a credential', async function(assert) {
+    let { path, scope, role } = await createRole();
+    await credentialsPage.visit({ backend: path, scope, role });
+    assert.ok(credentialsPage.isEmpty, 'renders empty creds page');
+    await credentialsPage.generateCredentialsLink();
+    assert.equal(
+      currentURL(),
+      `/vault/secrets/${path}/kmip/scopes/${scope}/roles/${role}/credentials/generate`,
+      'navigates to generate credentials'
+    );
+    await credentialsPage.submit();
+    assert.equal(
+      currentRouteName(),
+      'vault.cluster.secrets.backend.kmip.credentials.show',
+      'generate redirects to the show page'
+    );
+    await credentialsPage.backToRoleLink();
+
+    assert.equal(credentialsPage.listItemLinks.length, 1, 'renders a single credential');
+
+    // revoke the credentials
```

Comment:
this is just a style preference, but is it possible to separate the create and delete tests for scopes, roles, and credentials into separate tests (i.e. testing create and delete separately for each entity)? the tests are starting to look pretty long which can be harder to debug later. if it's too much work to set up each test though then i don't think it' a big deal at all though. 🤷‍♀

Reply:
Yep, I'll look at divvying these up more.